### ISSUE TITLE
Should browser-monkey really do all those assertions?

### ISSUE THREAD

**Issue by joshski** [featurist/browser-monkey#74]
> Browser Monkey is a DOM *assertion* library

Is it? ....really?

I think it's core strength is actually finding and manipulating elements in a DOM. To back up this statement, consider the fact that the [first example in the readme](https://github.com/featurist/browser-monkey#example) does no assertions!

I feel like maybe assertions should be left to the user. So perhaps I would rather express this:

```js
await browser.find('ul.monkey-species li').shouldHave({ text: [
  'Olive Baboon',
  'Patas Monkey']
})
```

...as this:

```js
assert.deepEqual(await browser.find('ul.monkey-species li').text(), [
  'Olive Baboon',
  'Patas Monkey']
)
```

As a result:
* I can use whatever assertion library I want
* My tests throw assertion exceptions, instead of some framework
* The browser-monkey API has a smaller surface area, so it's easier to learn and maintain

What do you think?

**Comment by refractalize** [comment]
major problem here is that browser monkey retries assertions until they pass (or until timeout) - if we can get this to work with custom assertions then i'm in favour ;)

**Comment by joshski** [comment]
Yeah, good point! Maybe there's not much difference when you factor that stuff in. The way this works in capybara is that the driver answers a query, which eventually returns false, so:

```js
async browser.exists(options) // => returns false
```

...it's then up to the assertion framework to assert it exists or describe the failure, e.g:

```js
assert(await browser.exists(options), `Expected element to exist with ${options}`)
```

**Comment by joshski** [comment]
FWIW I think a wise strategy when waiting for something to exist, is wait for the smallest piece of evidence that it exists. In the example above, I think you generally want to be waiting for `ul.monkey-species` to exist, then synchronously asserting on the `li` items inside it. There's a subtle difference between that and waiting for a more complex thing (such as the list _with_ the items in it) but it's an important difference in my mind. Mixing the assertions with the finding makes it difficult to know exactly what the test is doing IMO.

**Comment by joshski** [comment]
Also, you know.... perhaps the retrying isn't really browser-monkey's responsibility either 😲 

```js
assert.eventuallyDeepEqual(browser.find('ul.monkey-species li').text(), [
  'Olive Baboon',
  'Patas Monkey']
)
```
